src/rmarkdown/preview.ts

```diff
@@ -182,6 +182,11 @@ export class RMarkdownPreviewManager {
         const fileUri = uri ?? vscode.window.activeTextEditor.document.uri;
         const fileName = fileUri.fsPath.substring(fileUri.fsPath.lastIndexOf(path.sep) + 1);
         const currentViewColumn: vscode.ViewColumn = vscode.window.activeTextEditor?.viewColumn ?? vscode.ViewColumn.Active ?? vscode.ViewColumn.One;
+        // abort if the file is untitled
+        if (!uri && vscode.window.activeTextEditor.document.isUntitled) {
+            void vscode.window.showWarningMessage('Cannot knit an untitled file. Please save the document and try again.');
```

Comment:
Instead of show warning message, maybe we should execute command `workbench.action.files.save`?

Reply:
Or disable the preview command when the current file is not an RMarkdown file? That's how the integrated markdown preview behaves.

Else, would it work to copy the file contents to a tempfile (in the background) and still render the preview for untitled files?

Reply:
An untitled document could be assigned a language. If you create an untitled document and choose `md` language, then the markdown preview will work.

Reply:
This reminds me that should we always save document before rendering?

Reply:
Personally, I would prefer this. This would be also in line with other extensions like latex-workshop which saves at compilation.

Reply:
> An untitled document could be assigned a language.

Ah ok, didn't know that, thanks